Sources/ServiceLifecycle/ServiceGroup.swift

```diff
@@ -447,18 +481,31 @@ public actor ServiceGroup: Sendable {
         self.logger.debug(
             "Service lifecycle ended"
         )
+        cancellationTimeoutTask?.cancel()
         try result.get()
     }
 
     private func shutdownGracefully(
         services: [ServiceGroupConfiguration.ServiceConfiguration?],
+        cancellationTimeoutTask: inout Task<Void, Never>?,
         group: inout ThrowingTaskGroup<ChildTaskResult, Error>,
         gracefulShutdownManagers: [GracefulShutdownManager]
     ) async throws {
         guard case .running = self.state else {
             fatalError("Unexpected state")
         }
 
+        if #available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *), let maximumGracefulShutdownDuration = self.maximumGracefulShutdownDuration {
+            group.addTask {
+                try await Task.sleep(for: Duration(
+                    secondsComponent: maximumGracefulShutdownDuration.secondsComponent,
+                    attosecondsComponent: maximumGracefulShutdownDuration.attosecondsComponent
+                ))
```

Comment:
I think `maximumGracefulShutdownDuration` is already a `Duration` here, no need to convert again